src/commands/build/wranglerjs/mod.rs

```diff
@@ -17,32 +17,75 @@ use std::iter;
-    let mut rng = thread_rng();
-    iter::repeat(())
-        .map(|()| rng.sample(Alphanumeric))
-        .take(n)
-        .collect()
-}
+use crate::settings::project::Project;
+
+use crate::terminal::message;
 
 // Run the underlying {wranglerjs} executable.
 //
 // In Rust we create a virtual file, pass the pass to {wranglerjs}, run the
 // executable and wait for completion. The file will receive the a serialized
 // {WranglerjsOutput} struct.
 // Note that the ability to pass a fd is platform-specific
-pub fn run_build(
-    wranglerjs_path: PathBuf,
-    wasm_pack_path: PathBuf,
-    webpack_config_path: PathBuf,
-    bundle: &Bundle,
-) -> Result<WranglerjsOutput, failure::Error> {
+pub fn run_build(cache: &Cache, project: &Project) -> Result<(), failure::Error> {
+    let (mut command, temp_file, bundle) = setup_build(cache, project)?;
+
+    info!("Running {:?}", command);
+
+    let status = command.status()?;
+    info!("it ran");
+    let output = fs::read_to_string(temp_file.clone()).expect("could not retrieve ouput");
+    fs::remove_file(temp_file)?;
+
+    if status.success() {
+        let wranglerjs_output: WranglerjsOutput =
+            serde_json::from_str(&output).expect("could not parse wranglerjs output");
+
+        if wranglerjs_output.has_errors() {
+            message::user_error(&format!("{}", wranglerjs_output.get_errors()));
+            failure::bail!("Webpack returned an error");
+        }
+
+        bundle
+            .write(&wranglerjs_output)
+            .expect("could not write bundle to disk");
+
+        let mut msg = format!(
+            "Built successfully, script size is {}",
+            wranglerjs_output.script_size()
+        );
+        if bundle.has_wasm() {
+            msg = format!("{} and Wasm size is {}", msg, wranglerjs_output.wasm_size());
+        }
+        message::success(&msg);
+        Ok(())
+    } else {
+        failure::bail!("failed to execute `{:?}`: exited with {}", command, status)
+    }
+}
+
+//setup a build to run wranglerjs, return the command, the ipc temp file, and the bundle
+fn setup_build(
+    cache: &Cache,
+    project: &Project,
+) -> Result<(Command, PathBuf, Bundle), failure::Error> {
+    for tool in &["node", "npm"] {
+        env_dep_installed(tool)?;
+    }
+
+    let current_dir = env::current_dir()?;
+    run_npm_install(current_dir).expect("could not run `npm install`");
+
     let node = which::which("node").unwrap();
     let mut command = Command::new(node);
+    let wranglerjs_path = install(cache).expect("could not install wranglerjs");
     command.arg(wranglerjs_path);
+
+    //put path to our wasm_pack as env variable so wasm-pack-plugin can utilize it
+    let wasm_pack_path = install::install("wasm-pack", "rustwasm", cache)?.binary("wasm-pack")?;
     command.env("WASM_PACK_PATH", wasm_pack_path);
 
-    // create temp file for special {wranglerjs} IPC.
+    //create a temp file to communicate with wranglerjs
```

Comment:
IPC stands for Inter-process communication, we are using a file between these two processes. I feel like we should keep that comment. We can rephrase it if it's not clear to you.

Reply:
```suggestion
    // create a temp file for IPC with the wranglerjs process
```